### PR TITLE
Fix ccache for MacOS

### PR DESCRIPTION
**Pull request by NabsiYa**
This should be the fix for ccache on MacOS CI builds it seemed to work when I tested it.

### CODE REVIEW

**Comment by NabsiYa** [comment]
Didn't solve the problem.

**Comment by Macdu** [comment]
From the ccache stat at the end of the MacOS build, it actually looks like it worked

**Comment by NabsiYa** [comment]
Interesting... I may then reopen this pr.

File: .github/workflows/c-cpp.yml [hunk 1]
```diff
@@ -70,6 +74,7 @@ jobs:
           key: cache-${{ matrix.os }}-${{ matrix.config }}-${{ github.sha }}
           restore-keys: |
             cache-${{ matrix.os }}-${{ matrix.config }}-
+        if: matrix.os == 'ubuntu-latest'
```
Comment (Macdu) [inline .github/workflows/c-cpp.yml:77]:
```suggestion
        if: matrix.os != 'macos-latest'
```

The Windows CI also uses the cache action for the vcpkg build cache.

**Comment by NabsiYa** [comment]
Looks good to me.

**Comment by NabsiYa** [comment]
I probably should put everything into one commit?

File: .github/workflows/c-cpp.yml (hunk 1 above)
Comment (Macdu) [reply .github/workflows/c-cpp.yml:77]: You modified the wrong line.

**Comment by Macdu** [comment]
Not necessary, I can "squash and merge" to only have one commit in the end.

**Comment by NabsiYa** [comment]
Ok, now it should be fine... Let's wait till the CI is done.

**Pull request closed without merging**